web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx

```diff
@@ -174,6 +178,26 @@ const BuyCreditsModal: React.FC<BuyCreditsModalProps> = ({
           validate={validationHandler}
           onSubmit={async (values, { setSubmitting }) => {
             setSubmitting(true);
+            if (location.pathname.includes('projects')) {
```

Comment:
this is so that we only send the `buy2` event from the projects page (which i believe is the desired behavior). without this, the `buy2` event gets sent from the `storefront` page as well.

Reply:
Actually, the modal is not only used on the projects page but also on the credit and home pages.
Should we reverse the condition to cover all pages that use this modal ?

```suggestion
            if (!location.pathname.includes('storefront')) {
```

Reply:
a diagram i made on monday that maybe help here... the `BuyCreditsModal` is being imported in 3 places directly:
* `ProjectDetails.TransactionModals.tsx` (I think it's legacy, associated with testProject)
* `Storefront.tsx`
* `BuySellOrderFlow.tsx` (it is a wrapper of the form modal and the other modals of success, error, etc). This component is imported in 6 other places (2 of them being the project details page again)

![image](https://user-images.githubusercontent.com/5287737/195331464-2cb3327e-3c8e-4e64-8c73-1b012158db88.png)

Reply:
@victordelval [see here](https://github.com/regen-network/regen-web/pull/1188#issuecomment-1270265241) for my replies to Florent, somehow i didn't see a reply dialogue for this thread when i was reading the reviews.

i think this diagram would be helpful when we are looking to do some refactoring, it's worth adding it to a related issue if we have one or creating a new one, thanks for sharing it though, it does help to see it.